lib/cocoapods/user_interface.rb

```diff
@@ -331,13 +337,18 @@ def puts(message = '')
       #        The message to print.
       #
       def print(message)
-        STDOUT.print(message) unless config.silent?
+        return if config.silent?
+        begin
+          (io || STDOUT).print(message)
+        rescue Errno::EPIPE
+          exit 0
+        end
       end
 
       # gets input from $stdin
       #
       def gets
-        $stdin.gets
+        (io || $stdin).gets
```

Comment:
Makes sense
On Sat, 14 Nov 2015 at 19:50, Samuel E. Giddins <notifications@github.com>
wrote:

> In lib/cocoapods/user_interface.rb
> <https://github.com/CocoaPods/CocoaPods/pull/4519#discussion_r44861435>:
>
> >        end
> >
> >        # gets input from $stdin
> >        #
> >        def gets
> > -        $stdin.gets
> > +        (io || $stdin).gets
>
> hmm, since most pipes used here won't be bidirectional, let's keep this as
> stdin for now?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CocoaPods/CocoaPods/pull/4519/files#r44861435>.
>